Expose the stored subsystem name via the subsystem handle

girstenbrei opened the pull request:
Closes #97

Finomnis commented:
Would you mind adding a test for it as well?

Finomnis commented:
Ideally, where you test two levels of meats subsystems to make sure the name gets properly constructed

Finomnis commented:
Would mind extending the test to check the nested-ness of nested subsystems? :) sorry to be pedantic :D

girstenbrei commented:
Not minding at all, no problem:

I added an integration test (is the `integration_test.rs` the preferred location? It looked like a good candidate) with 2 nested subsystems and testing their respective generated names and that the nest correctly.

I also added a small note in the original `name()` fn regarding how names are constructed. I think this should be documented somewhere, I just don't know if this is the correct location. Maybe the `SubsystemBuilder::new` would be better, what do you think?

> Would mind extending the test to check the nested-ness of nested subsystems? :) sorry to be pedantic :D

I'm not quite sure what you mean, sorry: You want to test that nested subsystems are actually nested? And no worries about being pedantic, let's do it right.

Finomnis commented:
That's exactly what I meant, that's perfect :)
